feat(llmobs): added eval tracing explanation (#38218)
DataDog/documentation: content/en/llm_observability/evaluations/custom_llm_as_a_judge_evaluations/_index.md

### Configure the prompt

1. In Datadog, navigate to the Agent Observability [Evaluations page][1]. Select {{< ui >}}Create Evaluation{{< /ui >}}, then select {{< ui >}}Create your own{{< /ui >}}.

>    {{< img src="llm_observability/evaluations/EvalConfig_LLMO_1.png" alt="The Agent Observability Evaluation page after selecting Create Evaluation ." style="width:100%;" >}}

Reviewer: ```suggestion
   {{< img src="llm_observability/evaluations/EvalConfig_LLMO_1.png" alt="The Agent Observability Evaluations page after selecting Create Evaluation." style="width:100%;" >}}
```
Alt text has a space before the period, and this page is called the "Evaluations" page (plural) elsewhere.

> 1. To enable tracing for evaluations, select the {{< ui >}} Tracing Disabled {{< /ui >}} button and select the toggle to enable tracing. Evaluation traces will appear under `datadog-evaluations` when this evaluation is run, allowing greater visibility into your evaluations. **Note**: Enabling tracing increases the number of billed spans sent to Datadog. 

Reviewer: ```suggestion
1. To enable tracing for evaluations, click the {{< ui >}}Tracing Disabled{{< /ui >}} button, then select the {{< ui >}}Trace Evaluations{{ </ui> }} toggle to enable tracing. When this evaluation runs, its traces appear under `datadog-evaluations`, giving you greater visibility into your evaluations. **Note**: Enabling tracing increases the number of billed spans sent to Datadog. 
```

    {{< img src="llm_observability/evaluations/evaluation_tracing_enabled.png" alt="Trace Evaluations enabled after the toggle to enable evaluation tracing has been selected." >}}
1. Provide a clear, descriptive {{< ui >}}evaluation name{{< /ui >}} (for example, `factuality-check` or `tone-eval`). You can use this name when querying evaluation results. The name must be unique within your application.

> 1. Select the {{< ui >}}Model{{< /ui >}} to configure the model. 

Reviewer: ```suggestion
1. Configure the model:
```

>     1. Select the {{< ui >}}Accout{{< /ui>}}  drop-down menu to select the LLM provider and corresponding account to use for your LLM judge. To connect a new account, see [connect an LLM provider][2].

Reviewer: ```suggestion
    1. Click the model name next to {{< ui >}}Account{{< /ui >}} to open the configuration window.
    1. Use the {{< ui >}}Account{{< /ui >}} dropdown menu to select the LLM provider and corresponding account to use for your LLM judge. To connect a new account, see [connect an LLM provider][2].
```

Author: This is incorrect. After selecting the model a window pops up from which in the window they select the account from the drop-down menu.

Author: Actually this should be the case for all cases unless there are not any accounts configured. Feel free to disregard my previous comment :)

        - If you select an {{< ui >}}Amazon Bedrock{{< /ui >}} account, choose a region the account is configured for. You can then select a model name or provide the inference profile ARN.
        - If you select a {{< ui >}}Vertex{{< /ui >}} account, choose a project and location.

>     1. Use the {{< ui >}}Model{{< /ui >}} drop-down menu to select a model to use for your LLM judge.

Reviewer: ```suggestion
    1. Use the {{< ui >}}Model{{< /ui >}} dropdown menu to select a model.
```
"drop-down" -> "dropdown" for consistency with the docs convention.

> 1. In {{< ui >}}Runs On{{< /ui >}}, select the application you want to evaluate, what you want to evaluate on (span, trace, or session), and the sampling rate. You can add more filtering criteria by clicking the button to the right of sampling.

Reviewer: ```suggestion
1. In {{< ui >}}Runs On{{< /ui >}}, select the application you want to evaluate, what you want to evaluate on (span, trace, or session), and the sampling rate. You can add more filtering criteria by selecting the button to the right of the sampling rate.
```

> 1. In the {{< ui >}}Template{{< /ui >}} section, use the drop-down menu:

Reviewer: ```suggestion
1. In the {{< ui >}}Template{{< /ui >}} section, use the dropdown menu:
```

It feels a little weird to have Template at this step in the procedure, given that it's the first field in the UI. Is there a reason you want it here vs. at the start?

Author: I have it here because typically the users select the template they want to create before landing on the eval configuration page, so if they were to change the template, it would be when they are modifying the system prompt.

```
{{% /collapse-content %}}

> 8. In the {{< ui >}}User Prompt{{< /ui >}} field, specify what parts of the span, trace, or session to evaluate by adding variables. You can add any span attribute, such as Span Input (`{{span_input}}`), Output (`{{span_output}}`), or any other span field. For trace-scoped evaluations, use `{{spans...}}` paths to read across spans; for session-scoped evaluations, use `{{traces...}}` paths to read across traces. See [Prompt Templating][15] for the full reference. To freely edit the user prompt, select the user prompt and edit the text.

Reviewer: ```suggestion
8. In the {{< ui >}}User Prompt{{< /ui >}} field, specify what parts of the span, trace, or session to evaluate by adding variables. You can add any span attribute, such as Span Input (`{{span_input}}`), Output (`{{span_output}}`), or any other span field. For trace-scoped evaluations, use `{{spans...}}` paths to read across spans; for session-scoped evaluations, use `{{traces...}}` paths to read across traces. See [Prompt Templating][15] for the full reference. To edit the user prompt directly, select it and edit the text.
```
Minor: "freely" is filler and the final sentence was slightly circular.

Reviewer: @jennm I still prefer adding this edit but it's up to you! Approving now

Author: I think I just added it! I didn't add it previously because it was hidden

   You may also use the panel on the right ({{< ui >}}Filtered Spans{{< /ui >}} in span scope, {{< ui >}}Filtered Traces{{< /ui >}} in trace scope, {{< ui >}}Filtered Sessions{{< /ui >}} in session scope) to add span data as a variable:

>    1. Choose an account and an application so that spans/traces/sessions show up on the right.

Reviewer: ```suggestion
   1. Choose an account and an application so that spans, traces, or sessions show up on the right.
```

I don't see where the user would choose an account and application? 

<img width="427" height="760" alt="Image" src="https://github.com/user-attachments/assets/5ef6df2a-eed9-4b6c-8a27-c70f25f0c961" />

Author: That's because I reduced the page to only show the filtered spans. Should the full screen be visible in this screenshot?

Reviewer: Oh no your screenshot is fine, I was just looking at it in the app myself to confirm the step and couldn't find it. Where do they choose an account and an application for this?

Author: They select an application here 
<img width="638" height="87" alt="Screenshot 2026-07-17 at 1 33 36 AM" src="https://github.com/user-attachments/assets/2f4abfe2-d0b0-4d6b-b9b7-6572586c01d7" />

and the account here 
<img width="598" height="693" alt="Screenshot 2026-07-17 at 1 34 16 AM" src="https://github.com/user-attachments/assets/9873726e-5bee-484d-9c75-579347acb55c" />

Author: The config page is super dense, so totally understand struggling to find certain things :)

   2. Select one of the spans on the right to view its JSON.

>    3. Select the {{< ui >}}+{{< /ui >}} to add the JSON into your user prompt

Reviewer: ```suggestion
   3. Select {{< ui >}}+{{< /ui >}} to add the JSON to your user prompt.
```

      - `@name:agent.workflow AND env:prod` to filter by span name and tag
   - {{< ui >}}Sampling Rate{{< /ui >}}: (Optional) Apply sampling (for example, 10%) to control evaluation cost.

{{< img src="llm_observability/evaluations/evaluation_scope_1.png" alt="Configuring the evaluation scope." style="width:100%;" >}}

### Test and preview

The pane on the right shows {{< ui >}}Filtered Spans{{< /ui >}} (or traces) corresponding to the configured evaluation scope.

Select a span to show JSON data available for use in an evaluation. Then, click {{< ui >}}Test Evaluation{{< /ui >}} to pre-fill inputs to your evaluation with data from the span, and click {{< ui >}}Run{{< /ui >}} to test.

> To enable tracing for your evaluation, select the toggle so that Tracing Enabled is on.

Reviewer: Enabling tracing is now described in three places: the intro, the configuration step above (step 2), and here. This sentence largely repeats the configuration step. Consider removing it so there is a single source of truth for enabling tracing.
